src/button-group/button-group.js

```diff
@@ -97,7 +106,7 @@ export class ButtonGroupRoot extends React.Component<{|
                   this.childRefs[nextIndex].current.focus();
               }
             },
-            kind,
+            kind: computedKind,
```

Comment:
```suggestion
            kind: !kind && isSelected(selected, index) ? KIND.primary : kind,
```

Reply:
If kind is not specified and the button is selected change it to primary, otherwise fallback to the specified kind. This code may be incorrect because there may be a default value for kind, but it described the intention